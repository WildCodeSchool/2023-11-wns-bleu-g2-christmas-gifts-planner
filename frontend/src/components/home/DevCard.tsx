import { Avatar, Card, Center, Heading, Text } from "@chakra-ui/react";
import Link from "next/link";
import React from "react";
import { useTranslation } from "react-i18next";

export default function DevCard({
  name,
  link,
  color,
  role,
  description,
}: {
  name: string;
  link: string;
  color: string;
  role: string;
  description: string;
}) {
  const { t } = useTranslation();

  return (
    <Card
      key={name}
      maxWidth="15rem"
      p={4}
      textAlign="center"
      transition={"all 0.3 ease"}
      _hover={{
        boxShadow: "2xl",
        transform: "translateY(-2px)",
        backgroundColor: "whiteAlpha.600",
        _dark: { backgroundColor: "dark.surface20" },
      }}
    >
      <Link href={link}>
        <Center>
          <Avatar size="lg" name={name} mb={4} backgroundColor={color} />
        </Center>
        <Heading size="md" mb={2}>
          {t(name)}
        </Heading>
        <Text fontWeight="bold" mb={2}>
          {t(role)}
        </Text>
        <Text fontSize="sm" color="gray.600" _dark={{ color: "gray.200" }}>
          {t(description)}
        </Text>
      </Link>
    </Card>
  );
}
