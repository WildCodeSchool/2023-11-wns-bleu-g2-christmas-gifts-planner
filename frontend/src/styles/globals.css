@tailwind base;
@tailwind components;
@tailwind utilities;

* {
  box-sizing: border-box;
}

html,
body {
  height: 100dvh;
}

#__next {
  height: 100%;
}

.genericButton {
  border: solid 1px transparent;
  border-radius: 6px;
  padding: 6px;
  transition: all 0.2s;
}
.genericButton:hover {
  background-color: #f0f0f0;
}
