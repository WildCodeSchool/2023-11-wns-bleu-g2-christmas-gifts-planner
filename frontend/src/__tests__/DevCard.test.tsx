import { describe, it, expect } from "@jest/globals";
import { render, screen } from "@testing-library/react";
import DevCard from "@/components/home/DevCard";

describe("Test component", () => {
    it("renders correctly", () => {
        const view = render (
        <DevCard 
        name="testName"
        link="testLink"
        color="testColor"
        role="testRole"
        description="testDescription" 
        />
    );
        expect(view.baseElement).toMatchSnapshot();
    });
});