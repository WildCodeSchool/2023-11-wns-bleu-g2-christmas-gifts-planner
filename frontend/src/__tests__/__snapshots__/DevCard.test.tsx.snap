// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component renders correctly 1`] = `
<body>
  <div>
    <div
      class="chakra-card css-1g230f1"
    >
      <a
        href="testLink"
      >
        <div
          class="css-gmuwbf"
        >
          <span
            class="chakra-avatar css-1ahe2zj"
          >
            <div
              aria-label="testName"
              class="chakra-avatar__initials css-0"
              role="img"
            >
              t
            </div>
          </span>
        </div>
        <h2
          class="chakra-heading css-wlea3r"
        >
          testName
        </h2>
        <p
          class="chakra-text css-1lp9c1c"
        >
          testRole
        </p>
        <p
          class="chakra-text css-130t7v7"
        >
          testDescription
        </p>
      </a>
    </div>
  </div>
</body>
`;
